### PR TITLE
Updated Notifications

### PR DESCRIPTION
**Pull request by nicolelst**
- cherry picked updated thread/meetup/comment/user + manager classes from detailed_meetup_view
- added consumer functionality for notifications
- aesthtic improvements to notifications view

### CODE REVIEW

File: lib/view/tabs/community/community_tab.dart [hunk 1]
```diff
@@ -32,57 +25,58 @@ class _CommunityTabState extends State<CommunityTab> {
...
+            return IconBadge(
+              icon: const Icon(Icons.notifications),
+              itemCount: notifMgr.getNumUnreadNotifications(),
+              badgeColor: Colors.red,
+              itemColor: Colors.white,
+              hideZero: true,
+              onTap: () => Navigator.push( // TODO ROUTING: replace w go router to notif view
+                  context,
+                  MaterialPageRoute(
+                      builder: (context) => NotificationsView(currUserID: _testUserID))),
```
Comment (nicolelst) [inline lib/view/tabs/community/community_tab.dart]: missing routing

File: lib/view/tabs/community/notifications_view.dart [hunk 2]
```diff
@@ -26,40 +27,43 @@ class _NotificationsViewState extends State<NotificationsView> {
...
                           // TODO: replace with go_router
                           Navigator.push(
                               context,
                               MaterialPageRoute(
                                   builder: (context) => DetailedMeetupView(
-                                      MeetupsController.getMeetup(
-                                          n.meetupID))));
+                                      meetupID: n.meetupID,
+                                      currUserID: widget.currUserID)));
                         }
```
Comment (nicolelst) [inline lib/view/tabs/community/notifications_view.dart]: missing routing

File: lib/view/tabs/community/community_tab.dart [hunk 3]
```diff
@@ -17,13 +16,7 @@ class CommunityTab extends StatefulWidget {
...
 
 class _CommunityTabState extends State<CommunityTab> {
-  List<no.Notification> notifications =
-  []; // TODO: get notifications from controller
-
-  void _tapNotifs() {
-    Navigator.push(context,
-        MaterialPageRoute(builder: (context) => const NotificationsView()));
-  }
+  final String _testUserID = "1"; // TODO READ USER ID OF VIEWER FROM SOMEWHERE?
```
Comment (nicolelst) [inline lib/view/tabs/community/community_tab.dart]: not sure where to get the user ID of the current user?

Comment (RealDyllon) [reply lib/view/tabs/community/community_tab.dart]: from the userManager

File: lib/view/tabs/community/notifications_view.dart (hunk 2 above)
Comment (RealDyllon) [reply lib/view/tabs/community/notifications_view.dart]: fixed

File: lib/view/tabs/community/community_tab.dart (hunk 1 above)
Comment (RealDyllon) [reply lib/view/tabs/community/community_tab.dart]: fixed

**Pull request merged**